pytest_inmanta/plugin.py

```diff
@@ -300,6 +302,7 @@ def dryrun(self, resource, run_as_root=False):
     def deploy_resource(self, resource_type: str, status=const.ResourceState.deployed, run_as_root=False, **filter_args: dict):
         res = self.get_resource(resource_type, **filter_args)
         assert res is not None, "No resource found of given type and filter args"
+        res = self.check_serialization(res)
```

Comment:
why not add it to get_resource directly, such that test that use the resource, but not the standard deploy also get the correct resource?

Reply:
Sure thing